src/runtime/refiner.ts

```diff
@@ -1313,27 +1341,15 @@ export class KTExtracter {
         return `val ${KOTLIN_QUERY_ARGUMENT_NAME} = queryArgs as ${codeGenerator.typeFor(type)}`;
     };
 
-    const fieldNames = new Set<string>();
-    const filterTerms = [];
-    if (schema.refinement) {
-      [...schema.refinement.getFieldParams().keys()].forEach(name => fieldNames.add(name));
-      filterTerms.push(schema.refinement.toKTExpression(codeGenerator));
-    }
-    for (const field of Object.values(schema.fields)) {
-      if (field.refinement) {
-        [...field.refinement.getFieldParams().keys()].forEach(name => fieldNames.add(name));
-        filterTerms.push(field.refinement.toKTExpression(codeGenerator));
-      }
-    }
+    const fieldNames = [...schema.getFieldParams().keys()].map(genFieldAsLocal);
+    const queryArgNames = [...schema.getQueryParams()].map(genQueryArgAsLocal);
+    const defineLocals = [...fieldNames, ...queryArgNames].map(def => `${def}\n`).join('');
+
+    const filterTerms = schema.toKTExpression(codeGenerator);
 
-    const locals = [...fieldNames].map(genFieldAsLocal);
-    if (schema.refinement) {
-      const querysArgs = [...schema.refinement.getQueryParams()].map(genQueryArgAsLocal);
-      locals.push(...querysArgs);
-    }
     const expr = filterTerms.length > 0 ? `${filterTerms.join(' && ')}` : 'true';
 
-    return `${locals.map(x => `${x}\n`).join('')}${expr}`;
+    return `${defineLocals}${expr}`;
```

Comment:
To continue from the last review...
`return [...defineLocals, expr].join('\n');`
The definition of `defineLocals` wouldn't need the `map` or `join`.

Reply:
This last line could be: 
```
return [...fieldNames, ...queryArgNames, expr].join('\n');
```

Reply:
Done (with an intermediate variable `statements`)